### PR TITLE
enforce_home_region session setting

### PR DESCRIPTION
**Pull request by taroface**
DOC-4744
DOC-5485

- Document `enforce_home_region` session setting
- Clarify definition of "home region" across the docs 
- A couple opportunistic clarifications to table locality descriptions

### CODE REVIEW

**Comment by taroface** [comment]
> Do we need to define home or "home" ? We've hesitated to do this in the past cc @ajstorm

Yeah, I wondered about this. Just made a couple tweaks to foreground this term more, but without making it an italicized _concept_. Does this work better for us?

**Review by rytaft — COMMENTED** [review]
This is great!

> Reviewed 2 of 6 files at r1, 1 of 1 files at r4, 8 of 8 files at r5, 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/15591#-:-NGxzGYS9Jpx2_pqshC7:b9znln1)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dikshant, @msirek, and @taroface)

___
*[`_includes/v22.2/misc/session-vars.md` line 22 at r6](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxya2H7_p90myQE6rn:-NGxya2H7_p90myQE6ro:b-xsbbc8) ([raw file](https://github.com/cockroachdb/docs/blob/2043819077d6f275297164f586afdc0f77d393fd/_includes/v22.2/misc/session-vars.md#L22)):*
> ```Markdown
> | <a name="enable-insert-fast-path"></a> `enable_insert_fast_path` | Indicates whether CockroachDB will use a specialized execution operator for inserting into a table. We recommend leaving this setting `on`. | `on` | Yes | Yes |
> | <a name="enable-zigzag-join"></a> `enable_zigzag_join` | Indicates whether the [cost-based optimizer](cost-based-optimizer.html) will plan certain queries using a zig-zag merge join algorithm, which searches for the desired intersection by jumping back and forth between the indexes based on the fact that after constraining indexes, they share an ordering. | `on` | Yes | Yes |
> | <a name="enforce-home-region"></a> `enforce_home_region` | **New in v22.2:** If set to `on`, queries return an error and in some cases a suggested resolution if they cannot run entirely in their home region. This can occur if a query has no home region (for example, if it reads from different home regions in a [regional by row table](multiregion-overview.html#regional-by-row-tables)) or a query's home region differs from the [gateway](architecture/life-of-a-distributed-transaction.html#gateway) region. For more information about home regions, see [Table localities](multiregion-overview.html#table-localities).<br/><br/>Only tables with `ZONE` [survivability](when-to-use-zone-vs-region-survival-goals.html) can be scanned without error when this is enabled. | `off` | Yes | Yes |
> ```

Do we want to state that this setting is "best effort" and/or experimental? Or perhaps in "preview"? cc @msirek 
___
*[`_includes/v22.2/sql/regional-by-row-table-description.md` line 1 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxgSGl3hCLcPBkEP2I:-NGxgSGl3hCLcPBkEP2J:b-yqngpp) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/_includes/v22.2/sql/regional-by-row-table-description.md#L1)):*
> ```Markdown
> In a _regional by row_ table, individual rows are optimized for access from different home regions. Every row in a regional by row table has a hidden [`crdb_region` column](set-locality.html#crdb_region) that represents the row's home region. The `REGIONAL BY ROW` setting automatically divides a table and all of [its indexes](multiregion-overview.html#indexes-on-regional-by-row-tables) into [partitions](partitioning.html), with each partition optimized for access from a different region. Like [regional tables](multiregion-overview.html#regional-tables), _regional by row_ tables are optimized for access from a single region. However, that region is specified at the row level instead of applying to the whole table.
> ```

It's not required that the column is hidden or that it is called `crdb_region`, although this is the default. The only requirement is that there is one column of type `crdb_internal_region` that is used as the partitioning column.
___
*[`v22.2/cost-based-optimizer.md` line 232 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxkcfuAfAwbFv6O0k8:-NGxkcfuAfAwbFv6O0k9:b-p06kws) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/v22.2/cost-based-optimizer.md#L232)):*
> ```Markdown
> ## Control whether queries are limited to a single region
> 
> Although the optimizer prefers to [read from columns in local regions](#locality-optimized-search-in-multi-region-clusters) when possible, by default, it does not guarantee that any query will not visit a remote region. This can occur if a query has no [home region](multiregion-overview.html#table-localities) (for example, if it reads from different home regions in a [regional by row table](multiregion-overview.html#regional-by-row-tables)) or a query's home region differs from the [gateway](architecture/life-of-a-distributed-transaction.html#gateway) region.
> ```

"read from columns in local regions" -> "read from rows in local regions"
___
*[`v22.2/multiregion-overview.md` line 181 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxwhsrBHYNNPLcg69Q:-NGxwhsrBHYNNPLcg69R:b-4owbc0) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/v22.2/multiregion-overview.md#L181)):*
> ```Markdown
> ## Table localities
> 
> _Table locality_ determines how CockroachDB optimizes access to CockroachDB optimizes access to the table's data. Every table in a multi-region database has a "table locality setting" that configures one or more *home regions* at the table or row level. A table or row's home region is where the [leaseholder](replication-layer.html#leases) of its ranges is placed, along with a number of voting replicas determined by the [survival goal](when-to-use-zone-vs-region-survival-goals.html) of the database.
> ```

"optimizes access to CockroachDB optimizes access to" <- remove a few of those words :)
___
*[`v22.2/set-locality.md` line 82 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxxn6r7Wta8J7G0cYk:-NGxxn6r7Wta8J7G0cYl:b-k6g67l) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/v22.2/set-locality.md#L82)):*
> ```Markdown
> <a name="crdb_region"></a>
> 
> Every row in a regional by row table has a hidden `crdb_region` column that represents the row's [home region](multiregion-overview.html#table-localities). To see a row's home region, issue a statement like the following:
> ```

I see that this existing text already mentions the hidden `crdb_region` column... feel free to ignore my previous comment, or you could consider changing it in both places (maybe as simple as adding "by default")

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/15591)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm, @awoods187, @dikshant, @msirek, and @rytaft)

___
*[`_includes/v22.2/misc/session-vars.md` line 22 at r6](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxya2H7_p90myQE6rn:-NH07IyT47OjcqlZDP-0:bm7sdr5) ([raw file](https://github.com/cockroachdb/docs/blob/2043819077d6f275297164f586afdc0f77d393fd/_includes/v22.2/misc/session-vars.md#L22)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

Do we want to state that this setting is "best effort" and/or experimental? Or perhaps in "preview"? cc @msirek 
</blockquote></details>

I added the phrase "This feature is in preview. It is subject to change." Let me know if this requires more detail, @msirek @rytaft.
___
*[`_includes/v22.2/sql/global-table-description.md` line 1 at r2](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGXLMET5PjS0XshJgWX:-NH02EdD2l-0rpLRFVnP:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f5cf574ae4efed0b12b71d6a05fc1c75966e0a03/_includes/v22.2/sql/global-table-description.md#L1)):*
<details><summary><i>Previously, awoods187 (Andy Woods) wrote…</i></summary><blockquote>

Do we need to define home or "home" ? We've hesitated to do this in the past cc @ajstorm 

We've also used "region affinity" 
</blockquote></details>

Done.
___
*[`_includes/v22.2/sql/regional-by-row-table-description.md` line 1 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxgSGl3hCLcPBkEP2I:-NH02VHlDK5thPn378ev:b8f2s73) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/_includes/v22.2/sql/regional-by-row-table-description.md#L1)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

It's not required that the column is hidden or that it is called `crdb_region`, although this is the default. The only requirement is that there is one column of type `crdb_internal_region` that is used as the partitioning column.
</blockquote></details>

Thanks for clarifying; I've adjusted this text!
___
*[`v22.2/cost-based-optimizer.md` line 232 at r2](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGXLelg9EqNrJ440L2l:-NH04hbc5W_Kq3-Cw9Pl:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f5cf574ae4efed0b12b71d6a05fc1c75966e0a03/v22.2/cost-based-optimizer.md#L232)):*
<details><summary><i>Previously, awoods187 (Andy Woods) wrote…</i></summary><blockquote>

We definitely need to define home somewhere 
</blockquote></details>

Done.
___
*[`v22.2/cost-based-optimizer.md` line 232 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxkcfuAfAwbFv6O0k8:-NH04lbEBs_zB8UZaVAI:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/v22.2/cost-based-optimizer.md#L232)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

"read from columns in local regions" -> "read from rows in local regions"
</blockquote></details>

Done.
___
*[`v22.2/multiregion-overview.md` line 181 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxwhsrBHYNNPLcg69Q:-NH04rU631RnHoM_OsDZ:brpbssd) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/v22.2/multiregion-overview.md#L181)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

"optimizes access to CockroachDB optimizes access to" <- remove a few of those words :)
</blockquote></details>

Yikes! Fixed.
___
*[`v22.2/set-locality.md` line 82 at r5](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxxn6r7Wta8J7G0cYk:-NH04ulh4ubkn8ps9h9h:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/08b943286507f845b237d645949bd863cfbd7570/v22.2/set-locality.md#L82)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

I see that this existing text already mentions the hidden `crdb_region` column... feel free to ignore my previous comment, or you could consider changing it in both places (maybe as simple as adding "by default")
</blockquote></details>

Done.

**Review by rytaft — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/15591#-:-NH0kXj75ziSPEvUgC8c:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajstorm, @awoods187, @dikshant, and @msirek)

___
*[`_includes/v22.2/misc/session-vars.md` line 22 at r6](https://reviewable.io/reviews/cockroachdb/docs/15591#-NGxya2H7_p90myQE6rn:-NH0k87yApLw_4l72Y02:bb6ywmx) ([raw file](https://github.com/cockroachdb/docs/blob/2043819077d6f275297164f586afdc0f77d393fd/_includes/v22.2/misc/session-vars.md#L22)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

I added the phrase "This feature is in preview. It is subject to change." Let me know if this requires more detail, @msirek @rytaft.
</blockquote></details>

LGTM

File: v22.2/when-to-use-zone-vs-region-survival-goals.md [hunk 1]
```diff
@@ -11,8 +11,8 @@ A [_survival goal_](multiregion-overview.html#survival-goals) dictates how many
 
 Allowed survival goals include:
 
-- `ZONE` (default)
-- `REGION`
+- `ZONE` (default). Databases configured to survive zone failures have 3 [voting replicas](architecture/overview.html#architecture-replica) for every range, all in the [home region](multiregion-overview.html#table-localities). To support low-latency reads from other regions, one [non-voting replica](architecture/replication-layer.html#non-voting-replicas) is placed in each non-home region.
```
Comment (kathancox) [inline v22.2/when-to-use-zone-vs-region-survival-goals.md:14]: The periods after `ZONE` and `REGION` should be colons?

**Review by kathancox — APPROVED** [review]
LGTM! Just one typo I think!

**Pull request merged**